pkgdb/src/pkgdb/input.cc

```diff
@@ -72,12 +73,19 @@ PkgDbInput::init()
     }
 
   /* If the schema version is bad, delete the DB so it will be recreated. */
-  SqlVersions dbVersions = this->dbRO->getDbVersion();
-  if ( dbVersions.tables != sqlVersions.tables )
+  const ScrapeRules & scrapeRules  = getDefaultRules();
+  SqlVersions         dbVersions   = this->dbRO->getDbVersion();
+  ScrapeMeta          dbScrapeMeta = this->dbRO->getDbScrapeMeta();
+  if ( bool rulesMatch = ( dbScrapeMeta.rulesHash == scrapeRules.hashString() );
+       dbVersions.tables != sqlVersions.tables || ! rulesMatch )
     {
       nix::logger->log(
         nix::lvlTalkative,
-        nix::fmt( "Clearing outdated database '%s'", this->dbPath.string() ) );
+        nix::fmt( "Outdated database '%s'", this->dbPath.string() ) );
+      nix::logger->log( nix::lvlTalkative,
+                        nix::fmt( "Clearing due to %s",
+                                  rulesMatch ? "table schema being outdated"
+                                             : "scraping rules mismatch" ) );
       std::filesystem::remove( this->dbPath );
       PkgDb( this->getFlake()->lockedFlake, this->dbPath.string() );
```

Comment:
question: does this invalidate and re-scrape immediately or on the next run? These 2 lines are making me think it happens immediately but your comments make me think it happens on the next run

Reply:
That's a good question I should resolve.  That confused me as well, but I thought I was just missing something and followed the same pattern.  It behaves as if it invalidates it for the NEXT run (confirmed by tests), but it reads as though it's immediate.  Will look further.